packages/neon/neon_dashboard/lib/src/widgets/widget.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:neon/theme.dart';
+import 'package:neon/widgets.dart';
+import 'package:neon_dashboard/l10n/localizations.dart';
+import 'package:neon_dashboard/src/widgets/widget_item.dart';
+import 'package:nextcloud/dashboard.dart' as dashboard;
+
+class DashboardWidget extends StatelessWidget {
+  const DashboardWidget({
+    required this.widget,
+    required this.items,
+    super.key,
+  });
+
+  final dashboard.Widget widget;
+  final dashboard.WidgetItems? items;
+
+  @override
+  Widget build(final BuildContext context) {
+    final halfEmptyContentMessage = items?.halfEmptyContentMessage ?? '';
+    final emptyContentMessage = items?.emptyContentMessage ?? '';
+
+    return SizedBox(
+      width: 320,
+      height: 560,
+      child: Card(
+        child: InkWell(
+          onTap: widget.widgetUrl != null && widget.widgetUrl!.isNotEmpty ? () => context.go(widget.widgetUrl!) : null,
+          borderRadius: const BorderRadius.all(Radius.circular(12)),
+          child: ListView(
```

Comment:
Maybe just write a generator function where you can cleanly add multiple checks into a condition.
Or make this a `ListView.builder` that takes a builder function by default.

Reply:
I would keep it how it is now. There is no benefit in using a builder because all items will always be displayed at once and there won't be many at once anyway (it's limited to 7 by default).